oak-run/src/main/java/org/apache/jackrabbit/oak/run/GenerateFullGCCommand.java

```diff
@@ -0,0 +1,416 @@
+                            "Applies only if numberOfRuns is greater than 1.").withRequiredArg()
+                    .ofType(Integer.class).defaultsTo(60);
+        }
+
+        public GenerateFullGCOptions parse(String[] args) {
+            super.parse(args);
+            return this;
+        }
+
+        String getSubCmd() {
+            List<String> args = getOtherArgs();
+            if (args.size() > 0) {
+                return args.get(0);
+            }
+            return CMD_GENERATE;
+        }
+
+        public int getCreateGarbageNodesCount() {
+            return createGarbageNodesCount.value(options);
+        }
+
+        public int getGarbageNodesParentCount() {
+            return garbageNodesParentCount.value(options);
+        }
+
+        public int getGarbageType() {
+            return garbageType.value(options);
+        }
+
+        public int getNumberOfRuns() {
+            return numberOfRuns.value(options);
+        }
+
+        public int getGenerateIntervalSeconds() {
+            return generateIntervalSeconds.value(options);
+        }
+    }
+
+    public void execute(String... args) throws Exception {
+        Closer closer = Closer.create();
+        try {
+            execute(closer, args);
+        } catch (Throwable e) {
+            LOG.error("Command failed", e);
+            throw closer.rethrow(e);
+        } finally {
+            closer.close();
+        }
+    }
+
+    /**
+     * Method with passed closer is necessary in order to allow for unit tests to check the output of the command.
+     * It is the responsibility of the caller to close the closer.
+     *
+     * Returns the list of generated garbage base paths (under the garbage root node).
+     * @param closer
+     * @param args
+     * @throws Exception
+     */
+    public List<String> execute(Closer closer, String... args) throws Exception {
+        continuousRunIndex = 0;
+
+        List<String> generateBasePaths = new ArrayList<>();
+
+        GenerateFullGCOptions options = new GenerateFullGCOptions(USAGE).parse(args);
+        String subCmd = options.getSubCmd();
+
+        if (GenerateFullGCOptions.CMD_GENERATE.equals(subCmd)) {
+            if (options.getNumberOfRuns() > 1 && options.getGenerateIntervalSeconds() > 0) {
+                generateBasePaths.addAll(generateGarbageContinuously(options, closer));
+            } else {
+                generateBasePaths.add(generateGarbage(options, closer, 0));
+            }
+        } else if (GenerateFullGCOptions.CMD_CLEAN.equals(subCmd)) {
+            cleanGarbage(options, closer);
+        } else {
+            System.err.println("unknown revisions command: " + subCmd);
+        }
+
+        return generateBasePaths;
+    }
+
+    private List<String> generateGarbageContinuously(GenerateFullGCOptions options, Closer closer) throws IOException, Exception {
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
```

Comment:
Could you please close the executor once finished.
See https://github.com/shodaaan/jackrabbit-oak/blob/trunk/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStoreThrottlingMetricsUpdater.java#L58 for reference.

Reply:
Changed. Thank you for the suggestion.